util/Android/scrollable.ts

```diff
@@ -0,0 +1,120 @@
+import IView from '../../ui/view/view';
+import ListViewItem from '../../ui/listviewitem';
+import AndroidUnitConverter from '../../util/Android/unitconverter';
+import ListView from '../../ui/listview';
+import GridView from '../../ui/gridview';
+
+const NativeRecyclerView = requireClass('androidx.recyclerview.widget.RecyclerView');
+const NativeSwipeRefreshLayout = requireClass('androidx.swiperefreshlayout.widget.SwipeRefreshLayout');
+
+export default class Scrollable {
+  protected _android: any; /**TODO: Check out with Cenk */
+  onPullRefresh(): void {}
+  nativeInner: any;
+  nativeObject: any;
+  nativeDataAdapter: any;
+  private _overScrollMode = 0;
+  private _onGesture: (distances: { distanceX: number; distanceY: number }) => boolean;
+  private _onAttachedToWindow: (...args: any) => any;
+  private _onDetachedFromWindow: (...args: any) => any;
+  constructor() {
+    this.nativeObject.setOnRefreshListener(
+      NativeSwipeRefreshLayout.OnRefreshListener.implement({
+        onRefresh: () => {
+          this.onPullRefresh?.();
+        }
+      })
+    );
+
+    this.nativeInner.addOnItemTouchListener(
+      NativeRecyclerView.OnItemTouchListener.implement({
+        onInterceptTouchEvent: () => !(this as any).touchEnabled,
+        onRequestDisallowInterceptTouchEvent: () => {},
+        onTouchEvent: () => {}
+      })
+    );
+
+    const self = this;
+    const android = {
+      get onGesture() {
+        return self._onGesture;
+      },
+      set onGesture(value: Scrollable['_onGesture']) {
+        self._onGesture = value;
+        if (self._onGesture) {
+          self.nativeInner.setJsCallbacks({
+            onScrollGesture: (distanceX: number, distanceY: number) => {
+              const returnValue = self._onGesture?.({ distanceX: distanceX, distanceY: distanceY }) ?? true;
+              return !!returnValue;
+            }
+          });
+        } else {
+          self.nativeInner.setJsCallbacks(null);
+        }
+      },
+      get overScrollMode() {
+        return self._overScrollMode;
+      },
+      set overScrollMode(mode: Scrollable['_overScrollMode']) {
+        const nativeLayout = (self instanceof ListView || self instanceof GridView) ? self.nativeInner : self.nativeObject;
+        nativeLayout.setOverScrollMode(mode);
+        self._overScrollMode = mode;
+      },
+      saveInstanceState() {
+        const layoutManager = self instanceof GridView ? self.layoutManager : undefined
+        return {
+          nativeObject: (layoutManager as any)?.nativeObject.onSaveInstanceState()
+        }
+      },
+      restoreInstanceState(savedInstance: any) {
+        const layoutManager = self instanceof GridView ? self.layoutManager : undefined;
+        (layoutManager as any).nativeObject.onRestoreInstanceState(savedInstance.nativeObject);
```

Comment:
clear `as any`

Reply:
Then it will give type error until layoutManager is successfully transferred to Typescript